src/NuGet.Core/NuGet.CommandLine.XPlat/Utility/MSBuildAPIUtility.cs

```diff
@@ -743,14 +744,40 @@ internal static bool IsPackageReferenceProject(Project project)
                     if (matchingPackages.Any())
                     {
                         var topLevelPackage = matchingPackages.Single();
-                        InstalledPackageReference installedPackage;
+                        InstalledPackageReference installedPackage = default;
 
                         //If the package is not auto-referenced, get the version from the project file. Otherwise fall back on the assets file
                         if (!topLevelPackage.AutoReferenced)
                         {
                             try
                             { // In case proj and assets file are not in sync and some refs were deleted
-                                installedPackage = projPackages.Where(p => p.Name.Equals(topLevelPackage.Name, StringComparison.Ordinal)).First();
+                                var projectPackage = projectPackages.Where(p => p.Name.Equals(topLevelPackage.Name, StringComparison.Ordinal)).First();
+
+                                // If the project is using CPM and it's not using VersionOverride, get the version from Directory.Package.props file
```

Comment:
I feel like this is contradicting line 759 https://github.com/NuGet/NuGet.Client/pull/5350/files#diff-90c9045bb743a57a0e781f15088f964285ff4adf24e006ed4cd3a68d7e520674R759 which is handling verison overide cases.

Reply:
Removing lines 759 and its corresponding if block, VersionOverride is already handled like you pointed out below